Xamarin.Forms.Platform.Android/Renderers/MediaElementRenderer.cs

```diff
@@ -0,0 +1,380 @@
+
+				_view = new FormsVideoView(Context);
+				SetNativeControl(new FrameLayout(Context));
+
+				_view.SetZOrderMediaOverlay(true);
+				_view.SetOnCompletionListener(this);
+				_view.SetOnPreparedListener(this);
+				_view.KeepScreenOn = Element.KeepScreenOn;
+
+				Control.AddView(_view);
+
+				_controller = new MediaController(Context);
+				_controller.SetAnchorView(this);
+				_controller.Visibility = Element.AreTransportControlsEnabled ? ViewStates.Visible : ViewStates.Gone;
+				_view.SetMediaController(_controller);
+
+				UpdateSource();
+			}
+		}
+
+		void UpdateSource()
+		{
+			if (Element.Source != null)
+			{
+				if (Element.Source.Scheme == null)
+				{
+					_view.SetVideoPath(Element.Source.AbsolutePath);
+				}
+				else if (Element.Source.Scheme == "ms-appx")
+				{
+					// video resources should be in the raw folder with Build Action set to AndroidResource
+					string uri = "android.resource://" + Context.PackageName + "/raw/" + Element.Source.LocalPath.Substring(1, Element.Source.LocalPath.LastIndexOf('.') - 1).ToLower();
+					_view.SetVideoURI(global::Android.Net.Uri.Parse(uri));
+				}
+				else if (Element.Source.Scheme == "ms-appdata")
+				{
+					string filePath = string.Empty;
+
+					if (Element.Source.LocalPath.StartsWith("/local"))
+					{
+						filePath = System.IO.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), Element.Source.LocalPath.Substring(7));
+					}
+					else if (Element.Source.LocalPath.StartsWith("/temp"))
+					{
+						filePath = System.IO.Path.Combine(System.IO.Path.GetTempPath(), Element.Source.LocalPath.Substring(6));
+					}
+					else
+					{
+						throw new ArgumentException("Invalid Uri", "Source");
+					}
+
+					_view.SetVideoPath(filePath);
+
+				}
+				else
+				{
+					if (Element.Source.IsFile)
+					{
+						_view.SetVideoPath(Element.Source.AbsolutePath);
+					}
+					else
+					{
+						_view.SetVideoURI(global::Android.Net.Uri.Parse(Element.Source.ToString()), Element.HttpHeaders);
+					}
+				}
+
+				if (Element.AutoPlay)
+				{
+					_view.Start();
+					Element.SendCurrentState(MediaElementState.Playing);
+				}
+
+			}
+			else
+			{
+				if (Element.CurrentState == MediaElementState.Playing || Element.CurrentState == MediaElementState.Buffering)
+				{
+					Element.Stop();
+				}
+			}
+		}
+
+
+
+		protected override void OnElementPropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+		{
+			switch (e.PropertyName)
+			{
+				case nameof(MediaElement.AreTransportControlsEnabled):
+					_controller.Visibility = Element.AreTransportControlsEnabled ? ViewStates.Visible : ViewStates.Gone;
+					break;
+
+				case nameof(MediaElement.Source):
+					UpdateSource();
+					break;
+
+				case nameof(MediaElement.CurrentState):
+					switch (Element.CurrentState)
+					{
+						case MediaElementState.Playing:
+							if (!_view.IsPlaying)
+							{
+								_view.Start();
+							}
```

Comment:
After hitting the gallery buttons `Play` > `Stop` > `Play`, the video freezes and cannot be played again.
But If you press `Play` > `Pause` > `Stop` > `Play`, the video continues to play.